python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -96,6 +96,21 @@ def _getSchemaKeys(self, schema):
         self.fluxFlagKey = schema[fluxPrefix + "flag"].asKey()
         self.fluxSigmaKey = schema[fluxPrefix + "fluxSigma"].asKey()
 
+        self.edgeKey = schema["base_PixelFlags_flag_edge"].asKey()
+        self.interpolatedCenterKey = schema["base_PixelFlags_flag_interpolatedCenter"].asKey()
+        self.saturatedKey = schema["base_PixelFlags_flag_saturated"].asKey()
+
+    def _isGood_vector(self, sourceCat):
+        test = ~sourceCat.get(self.saturatedKey) & \
+               ~sourceCat.get(self.interpolatedCenterKey) & \
+               ~sourceCat.get(self.edgeKey)
+        return test
```

Comment:
Changed it. The function now returns the array rather than saving a temporary variable.